packages/e2e-tests/specs/performance/site-editor.test.js

```diff
@@ -75,112 +79,115 @@ describe( 'Site Editor Performance', () => {
 		}, html );
 		await saveDraft();
 
-		id = await page.evaluate( () =>
+		postId = await page.evaluate( () =>
 			new URL( document.location ).searchParams.get( 'post' )
 		);
 	} );
 
 	afterAll( async () => {
+		const resultsFilename = basename( __filename, '.js' ) + '.results.json';
+		writeFileSync(
+			join( __dirname, resultsFilename ),
+			JSON.stringify( results, null, 2 )
+		);
+
 		await deleteAllTemplates( 'wp_template' );
 		await deleteAllTemplates( 'wp_template_part' );
 		await activateTheme( 'twentytwentyone' );
 	} );
 
-	beforeEach( async () => {
-		await visitSiteEditor( {
-			postId: id,
-			postType: 'page',
-		} );
-	} );
-
-	it( 'Loading', async () => {
-		const editorURL = await page.url();
-
-		// Number of sample measurements to take.
+	describe( 'Loading', () => {
+		// Number of measurements to take.
 		const samples = 3;
 		// Number of throwaway measurements to perform before recording samples.
-		// Having at least one helps ensure that caching quirks don't manifest in
-		// the results.
+		// Having at least one helps ensure that caching quirks don't manifest
+		// in the results.
 		const throwaway = 1;
+		const iterations = sequence( 1, samples + throwaway );
+
+		it.each( iterations )(
+			`trace large post loading durations (%i of ${ iterations.length })`,
+			async ( i ) => {
+				// Open the test page in Site Editor.
+				await visitSiteEditor( {
+					postId,
+					postType: 'page',
+				} );
+
+				// Wait for the first content block to render.
+				await canvas().waitForSelector(
+					'[data-type="core/post-content"] .wp-block'
```

Comment:
Wait for the first block **inside the post content** instead of the first block in the whole canvas. If we wait for the first block in canvas as before, it will immediately return the document `header` block and refresh the page for the next loading test iteration. That approach ignores the actual content and the time editor needs to actually load the large post that is the subject of the test here. WDYT, @youknowriad?

Reply:
That's a good question, personally, I think the previous check might be a better check for the site editor. The site editor might not even have a post content block.

Reply:
Should we not load any content for the Loading perf tests, then? I guess if we only measure "Site Editor Loading", not "Site Editor **Content** Loading" it doesn't make sense to load any content at all.

Reply:
It seems both could be valid options. I'd still keep the content and only check the first block like we do for the post editor. We do have more "dynamic/lazy" blocks in the site editor but a user can still interact with the template even if all these are not loaded.

Reply:
> We do have more "dynamic/lazy" blocks in the site editor but a user can still interact with the template even if all these are not loaded.

In the current case, the post content block is loading almost 1.5k blocks synchronously, which freezes the browser for ~5s until all the blocks are rendered (you can see it [in the attached video here](https://github.com/WordPress/gutenberg/pull/48138#discussion_r1113032311)). It can significantly impact, e.g., the `time to first block` measurements if we _don't make it in time_ and hit refresh **after** the post content starts loading. I guess what I'm trying to say is that I don't understand the point of loading such heavy content if we ultimately need to avoid it or face inconsistent results. Does that make sense?

Reply:
Agreed, that seems counter intuitive. I'd probably load a hundred blocks or so within the template but there's no point in having 1500 with the post content block in the site editor.

Reply:
That said, it might make sense to have 1500 blocks for other metrics like typing (where you want to see how typing is with a lot of blocks).

Reply:
Alright, let's leave this aspect for a follow-up PR since it should also involve the post editor perf tests: I've [reverted](https://github.com/WordPress/gutenberg/pull/48138/commits/5623f43bebc772ff9a8d3fdb630daf0b5e3d7942) the `firstBlock` measurement to just `.wp-block` as it was before. The measurements after this PR should stay the same, only the general stability and readability of the site editor perf tests gets an improvement in this PR. It's ready for another go! 🙌